react-effects/my-app/src/List.js

```diff
@@ -0,0 +1,45 @@
+/* eslint-disable no-unused-vars -- REMOVE ME */
+import { useEffect, useState } from 'react';
+import readItems from './read';
+
+export default function List() {
+  const [isLoading, setIsLoading] = useState(true);
+  const [items, setItems] = useState([]);
+  const [error, setError] = useState();
+
+  // Your code here:
+  //  - When the component mounts:
+  //    - Read the items using `readItems` and update state so the list displays
+  //    - Handle errors from `readItems`
+  async function fetchData() {
+    try {
+      const data = await readItems();
+      setItems(data);
+      setIsLoading(false);
+    } catch (err) {
+      setError(err);
```

Comment:
This has a problem. If there is an error, isLoading never gets set to false. So you display the loading message instead of the error.

At least, that's how it normally works. In our case, with Strict Mode turned on, the useEffect gets called twice and the second time the readItems call succeeds, so it sets isLoading to false. If you turn off Strict Mode then you never see the error message. Bottom line -- you need to fix this! You need to make sure isLoading gets set to false whether the call succeeds or errors.